vxquery-core/src/main/java/org/apache/vxquery/metadata/VXQueryIndexingOperatorDescriptor.java

```diff
@@ -176,21 +177,20 @@ public void nextFrame(ByteBuffer buffer) throws HyracksDataException {
                             throw new HyracksDataException(
                                     "Could not delete index in " + indexModifiedName + " " + e.getMessage());
                         }
-
                     } else if (functionCall
-                            .equals(BuiltinFunctions.FN_COLLECTION_FROM_INDEX_2.getFunctionIdentifier().getName())) {
+                            .equals(BuiltinFunctions.FN_COLLECTION_FROM_INDEX_1.getFunctionIdentifier().getName())) {
```

Comment:
I think we should try to clean up this file by making this function smaller. Let's create individual functions for each of the index tasks. So this long if statement will just call individual functions that have all the logic for delete, create, read, etc. Lets try to make the function fit into one screen. No more scrolling.